qtaf-testrail-plugin/src/test/java/de/qytera/testrail/event_subscriber/UploadTestsSubscriberTest.java

```diff
@@ -44,6 +45,26 @@ public void testInitialization() {
 
     }
 
+
+    @TestRail(caseId = "01", runId = "01")
+    public void testDummyRunIdAnnotated(){}
+
+    @TestRail(caseId = "01", runId = "")
+    public void testDummyRunIdEmptyAnnotation(){}
+    @Test(description = "Test getRunId")
+    public void testGetRunId() throws ClassNotFoundException, NoSuchMethodException {
+        UploadTestsSubscriber subscriber = new UploadTestsSubscriber();
+        Class<?> dummyClass = Class.forName("de.qytera.testrail.event_subscriber.UploadTestsSubscriberTest");
+
+        Assert.assertThrows(NullPointerException.class, ()->subscriber.getRunId(null));
+
+        TestRail runIdEmptyAnnotation = dummyClass.getMethod("testDummyRunIdEmptyAnnotation").getAnnotation(TestRail.class);
+        Assert.assertThrows(IllegalArgumentException.class, () -> subscriber.getRunId(runIdEmptyAnnotation));
+
+        TestRail runIdAnnotatedAnnotation = dummyClass.getMethod("testDummyRunIdAnnotated").getAnnotation(TestRail.class);
+        Assert.assertEquals(subscriber.getRunId(runIdAnnotatedAnnotation), "01");
+    }
+
```

Comment:
This test case is totally correct, but I would split it up into three test cases, for each assertion one test case. This makes it easier to find errors later if a test case fails.

Reply:
Yes that is understandable. I have adjusted it.